test/testapps/TestServer/Startup.cs

```diff
@@ -26,9 +35,20 @@ public void ConfigureServices(IServiceCollection services)
                         .AllowAnyOrigin()
                         .AllowAnyHeader()
                         .AllowAnyMethod()
+                        .AllowCredentials()
                         .WithExposedHeaders("MyCustomHeader");
                 });
             });
+
+            services.AddAntiforgery(options =>
```

Comment:
I'm guessing this is meant to be used from an E2E test somewhere, but there doesn't appear to be one included in the PR. Is there still something to be finished here?

Reply:
yes the e2e test is missing.

Reply:
I tried to add an e2e test but cookies aren't recieved when selenium runs the test. If I run the test manualy, it works, but not with selenium. The code is :  

``` csharp
        [Fact]
        public void CanPerformPostRequestWithFetchCredentials()
        {
            new SelectElement(Browser.FindElement(By.Id("request-credentials")))
                    .SelectByValue("include");
            IssueRequest("GET", "/api/person/xhrf");
            Assert.Equal("OK", _responseStatus.Text);
            var token = _responseBody.Text;
            AddRequestHeader("X-XSRF-TOKEN", token);

            IssueRequest("POST", "/api/person/xhrf", token);
            Assert.Equal("OK", _responseStatus.Text);
            Assert.Equal($"You posted: {token}", _responseBody.Text);
        }
``` 

Any idea ?

Reply:
I'm not sure about that. Maybe Selenium requires additional config to enable cookies.

Instead of doing an E2E test about XSRF tokens, would it be simpler for the E2E test to cover a simpler `fetch` option such as [`referrer`](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch)?

Reply:
yes it is